src/agent/src/rpc.rs

```diff
@@ -243,6 +245,32 @@ impl AgentService {
             }
         }
 
+        let linux = oci
+            .linux()
+            .as_ref()
+            .ok_or_else(|| anyhow!("Spec didn't contain linux field"))?;
+        if let Some(devices) = linux.devices() {
+            for specdev in devices.iter() {
+                let dev_major_minor = format!("{}:{}", specdev.major(), specdev.minor());
+
+                if specdev.path().as_path().to_str() == Some(TRUSTED_IMAGE_STORAGE_DEVICE) {
+                    let secure_storage_integrity =
+                        AGENT_CONFIG.secure_storage_integrity.to_string();
+                    info!(
+                        sl(),
+                        "trusted_store device major:min {}, enable data integrity {}",
+                        dev_major_minor,
+                        secure_storage_integrity
+                    );
+
+                    Command::new(INIT_TRUSTED_STORAGE)
+                        .args([dev_major_minor.as_str(), &secure_storage_integrity])
+                        .output()
+                        .expect("Failed to initialize confidential storage");
+                }
```

Comment:
`break` the `for` loop once trusted storage is initialized ;)

Reply:
Done.